### ISSUE TITLE
Doctests error output is strange (and doesn't match to regex)

### ISSUE THREAD

**Issue by TorkelE** [JuliaDocs/Documenter.jl#1681]
Slight continuation from: https://github.com/JuliaDocs/Documenter.jl/issues/1679, but I think the issue is different, so opened a new issue.

Basically, I want to test this code, which yields an error message:
```
'''jldoctest; filter = r".*"s
[1,2,asf]

# output
'''
```
by using `doctest(MyPackage)`
(essentially I am trying to build a regex to catch whenever the code error, or not, and this is some of the early stages with trying to figure out regexes for the doctests)

My hope is that this regex will just accept anything. However, it doesn't seem to be able to match an error. This yields a crash:
```
┌ Error: doctest failure in ~/.julia/dev/Catalyst/docs/src/tutorials/using_catalyst.md:23-27
│ 
│ ```jldoctest; filter = r".*"s
│ [1,2,asf]
│ 
│ # output
│ 
│ ```
│ 
│ Subexpression:
│ 
│ [1,2,asf]
│ 
│ Evaluated output:
│ 
│ 
│ 
│ Expected output:
│ 
│ 
│ 
│   diff = 
└ @ Documenter.DocTests ~/.julia/packages/Documenter/R2HVS/src/DocTests.jl:385
```
Basically, the evaluated output of the error seem to be nothing at all. Which doesn't match to the regex? I get similar problem (but with different expected output) if I try some non-empty expected output:

```
'''jldoctest; filter = r".*"s
[1,2,asf]

# output
3
'''
```
(just something random, which is not correct)
```
'''jldoctest; filter = r".*"s
[1,2,asf]

# output
ERROR: UndefVarError: asf not defined
Stacktrace:
 [1] top-level scope
   @ REPL[1]:1
'''
```
(the error message which the code produces)

**Comment by mortenpi** [comment]
I wonder if it is related to #1676?
